net/uacme/Makefile

```
@@ -0,0 +1,70 @@
+#
+# Copyright (C) 2019 Lucian Cristian <lucian.cristian@gmail.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=uacme
+PKG_VERSION:=1.0.15
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/ndilieto/uacme/tar.gz/upstream/$(PKG_VERSION)?
+PKG_HASH:=b22e001928dd80249d7245b0cb61281ed487f8f1bd337b1264a754acf347561d
+
+PKG_MAINTAINER:=Lucian Cristian <lucian.cristian@gmail.com>
+PKG_LICENSE:=GPL-3.0
+PKG_LICENSE_FILES:=COPYING
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-upstream-$(PKG_VERSION)
+PKG_INSTALL:=1
+PKG_BUILD_PARALLEL:=1
+
+PKG_CONFIG_DEPENDS:= \
+  CONFIG_LIBCURL_GNUTLS \
+  CONFIG_LIBCURL_MBEDTLS \
+  CONFIG_LIBCURL_OPENSSL \
+  CONFIG_LIBCURL_WOLFSSL
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/uacme
+  SECTION:=net
+  CATEGORY:=Network
+  DEPENDS:=+libcurl +LIBCURL_WOLFSSL:libmbedtls
```

```suggestion
  DEPENDS:=libcurl @!LIBCURL_NOSSL +LIBCURL_WOLFSSL:libmbedtls
```
We will have a problem when libcurl is built without SSL.  I'm assuming that uacme can't run without support for https URLs.   If I'm wrong then at least, we will have to deal with the `configure` options in the same way we dealt with wolfssl, to for the use of mbedtls.
If we are to depend on `@!LIBCURL_NOSSL`, we can't select `libcurl`, or we will have a circular dependency.  It's getting hard to not make this visible only when libcurl is already selected (the suggestion above does that).

Another alternative is just display a warning message using `Package/uacme/config` below.

---

> I'm assuming that uacme can't run without support for https URLs

Your assumption is correct. The ACME protocol transport is strictly https.

uacme will still build even if libcurl has no SSL, but will fail at runtime when it tries to connect to https URLs. It will print out whatever error message libcurl throws up, which should help troubleshooting.